x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -21,13 +21,14 @@
 public class IndexDeprecationChecks {
 
     static DeprecationIssue oldIndicesCheck(IndexMetadata indexMetadata) {
-        Version createdWith = indexMetadata.getCreationVersion();
-        if (createdWith.before(Version.V_7_0_0)) {
+        // TODO: this check needs to be revised. It's trivially true right now.
+        Version currentCompatibilityVersion = indexMetadata.getCompatibilityVersion();
```

Comment:
I think this is right. But would we end up with a data-check as well, for instance if an upgrade from 8 to 9 means we no longer support ES 5 data, would we then have two checks go here?

Reply:
Yeah, the policy for how long we want to provide support for older versions needs to be decoupled from the N-1 index compatibility policy. While the checks in this PR are now purely based on `getCompatibilityVersion`, in the future they would be based on both `getCompatibilityVersion` and `getCreationVersion` (e.g. we could have 9.x reject indices created in 5.x, even if `getCompatibilityVersion() == 8.x`).